module-system/sov-modules-macros/src/make_constants.rs

```diff
@@ -0,0 +1,12 @@
+use syn::{Ident, Type};
+
+use crate::manifest::Manifest;
+
+pub(crate) fn make_const(
+    field_ident: &Ident,
+    ty: &Type,
+    vis: syn::Visibility,
+    attrs: &[syn::Attribute],
+) -> Result<proc_macro2::TokenStream, syn::Error> {
+    Manifest::read_constants(field_ident)?.parse_constant(ty, field_ident, vis, attrs)
```

Comment:
nit: should we consider making it a function of an instance of `Manifest`? This way will work, but we will parse the `constants.json` file every time the macro is called, as opposed to having it parsed only once.

This is optional as making the macro expansion stateful isn't trivial. One alternative would be to create a `static ref MANIFEST: Option<Manifest>` that would read the file once and serve it many times.

Reply:
Going to hold off on this for now to keep the PR small. I'm not too worried about the performance, since the JSON files we're parsing here are small